ReconnectTest fails to create constructable-ignored class MerkleStateRoot

`ReconnectTest` fails in the latest `develop` with the following exceptions:

```
Caused by: com.swirlds.common.merkle.synchronization.utility.MerkleSynchronizationException: unable to construct object with class ID -8201042766723040486(0x8E300B0DFDAFBB1A)
	at com.swirlds.common.merkle.synchronization.views.LearnerPushMerkleTreeView.deserializeInternal(LearnerPushMerkleTreeView.java:248)
	at com.swirlds.common.merkle.synchronization.views.LearnerPushMerkleTreeView.deserializeInternal(LearnerPushMerkleTreeView.java:50)
	at com.swirlds.common.merkle.synchronization.task.InternalDataLesson.deserialize(InternalDataLesson.java:116)
	at com.swirlds.common.merkle.synchronization.task.Lesson.deserialize(Lesson.java:110)
	at com.swirlds.common.merkle.synchronization.streams.AsyncInputStream.run(AsyncInputStream.java:149)
	at com.swirlds.common.threading.pool.StandardWorkGroup.lambda$execute$1(StandardWorkGroup.java:153)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

The forementioned class ID corresponds to `MerkleStateRoot`. This class is annotated with @ConstructableIgnored, this is why it isn't registered in the constructable registry.

--- a/platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/reconnect/ReconnectTest.java
+++ b/platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/reconnect/ReconnectTest.java
@@ -23,6 +23,7 @@ import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import com.swirlds.base.time.Time;
+import com.swirlds.common.constructable.ClassConstructorPair;
 import com.swirlds.common.constructable.ConstructableRegistry;
 import com.swirlds.common.constructable.ConstructableRegistryException;
 import com.swirlds.common.context.PlatformContext;
@@ -38,12 +39,14 @@ import com.swirlds.platform.metrics.ReconnectMetrics;
 import com.swirlds.platform.network.Connection;
 import com.swirlds.platform.network.SocketConnection;
 import com.swirlds.platform.state.MerkleRoot;
+import com.swirlds.platform.state.MerkleStateRoot;
 import com.swirlds.platform.state.RandomSignedStateGenerator;
 import com.swirlds.platform.state.signed.ReservedSignedState;
 import com.swirlds.platform.state.signed.SignedState;
 import com.swirlds.platform.state.signed.SignedStateValidator;
 import com.swirlds.platform.system.address.Address;
 import com.swirlds.platform.system.address.AddressBook;
+import com.swirlds.platform.test.NoOpMerkleStateLifecycles;
 import com.swirlds.platform.test.fixtures.addressbook.RandomAddressBookBuilder;
 import java.io.IOException;
 import java.security.PublicKey;
@@ -83,6 +86,9 @@ final class ReconnectTest {
         registry.registerConstructables("com.swirlds.common");
         registry.registerConstructables("com.swirlds.platform.state");
         registry.registerConstructables("com.swirlds.platform.state.signed");
+        registry.registerConstructables("com.swirlds.platform.system");
+        registry.registerConstructable(new ClassConstructorPair(
+                MerkleStateRoot.class, () -> new MerkleStateRoot(new NoOpMerkleStateLifecycles())));
     }
 
     @Test

